### PR TITLE
bug 1949306: add deprecated CRD manifest

### PR DESCRIPTION
**Pull request by deads2k**
simple rebase of https://github.com/openshift/cluster-kube-apiserver-operator/pull/1097 since normal work hours are over

### CODE REVIEW

File: bindata/v4.1.0/kube-apiserver/apiserver.openshift.io_deprecatedapirequests.yaml [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+    kind: DeprecatedAPIRequest
+    listKind: DeprecatedAPIRequestList
+    plural: deprecatedapirequests
+    singular: deprecatedapirequest
+  scope: Cluster
+  versions:
+  - name: v1
+    schema:
+      openAPIV3Schema:
+        description: DeprecatedAPIRequest tracts requests made to a deprecated API.
```
Comment (mfojtik) [inline bindata/v4.1.0/kube-apiserver/apiserver.openshift.io_deprecatedapirequests.yaml:20]: tracks ?

File: bindata/v4.1.0/kube-apiserver/apiserver.openshift.io_deprecatedapirequests.yaml [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: spec defines the characteristics of the resource.
+            type: object
+            properties:
+              removedRelease:
```
Comment (mfojtik) [inline bindata/v4.1.0/kube-apiserver/apiserver.openshift.io_deprecatedapirequests.yaml:43]: nit: `removedInRelease` reads better

File: bindata/v4.1.0/kube-apiserver/apiserver.openshift.io_deprecatedapirequests.yaml [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+                description: removedRelease is when the API will be removed.
+                type: string
+                maxLength: 64
+                minLength: 3
+                pattern: ^[0-9][0-9]*\.[0-9][0-9]*$
```
Comment (mfojtik) [inline bindata/v4.1.0/kube-apiserver/apiserver.openshift.io_deprecatedapirequests.yaml:48]: this allows `99.99` right? so the max length here is 5 ?

Comment (deads2k) [reply bindata/v4.1.0/kube-apiserver/apiserver.openshift.io_deprecatedapirequests.yaml:48]: pretty sure the * allows as many as you like.

**Comment by deads2k** [comment]
The job runs are failing after passes of the tests themselves

```
Gathering console logs for i-05b4a73e34ca258fd
'ascii' codec can't encode character '\u2026' in position 13825: ordinal not in range(128)
{"component":"entrypoint","error":"wrapped process failed: exit status 1","file":"prow/entrypoint/run.go:80","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2021-04-14T13:29:45Z"}
error: failed to execute wrapped command: exit status 1
```

Which is a known problem during collection, but not impacting the tests.  See https://coreos.slack.com/archives/CBN38N3MW/p1618410403100400

/override ci/prow/e2e-aws-serial
/override ci/prow/e2e-upgrade
/override ci/prow/e2e-aws
/override ci/prow/e2e-aws-operator

passing job runs are
1. https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-kube-apiserver-operator/1102/pull-ci-openshift-cluster-kube-apiserver-operator-master-e2e-aws-serial/1382254558310305792
2. https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-kube-apiserver-operator/1102/pull-ci-openshift-cluster-kube-apiserver-operator-master-e2e-upgrade/1382293842832658432
3. https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-kube-apiserver-operator/1102/pull-ci-openshift-cluster-kube-apiserver-operator-master-e2e-aws/1382303390985359360

File: bindata/v4.1.0/kube-apiserver/apiserver.openshift.io_deprecatedapirequests.yaml (hunk 3 above)
Comment (sanchezl) [reply bindata/v4.1.0/kube-apiserver/apiserver.openshift.io_deprecatedapirequests.yaml:48]: At least one digit on each side.

**Pull request merged**